File: Java/src/LongestPalindromicSubstringTest.java
import org.junit.Assert;
import org.junit.Test;

public class LongestPalindromicSubstringTest {

    @Test
    public void testLongestPalindrome() {
        LongestPalindromicSubstring l = new LongestPalindromicSubstring();

        String s = l.longestPalindrome("babad");
        Assert.assertTrue(s.equals("bab") || s.equals("aba"));

        Assert.assertEquals(l.longestPalindrome("cbbd"), "bb");

        Assert.assertEquals(l.longestPalindrome("babaddtattarrattatddetartrateedredividerb"), "ddtattarrattatdd");
    }

}

File: Kotlin/src/LongestPalindromicSubstringTest.kt
import org.junit.Test

class LongestPalindromicSubstringTest {

    @Test
    fun testLongestPalindrome() {
        val l = LongestPalindromicSubstring()

        val s0 = l.longestPalindrome("babad")
        assert(s0 == "bab" || s0 == "aba")

        assert(l.longestPalindrome("cbbd") == "bb")

        assert(l.longestPalindrome("babaddtattarrattatddetartrateedredividerb") == "ddtattarrattatdd")
    }

}